Source/Util/MapScanner.c

```diff
@@ -0,0 +1,159 @@
+// Copyright CircumScriptor and DarkNeutrino 2021
+#include "MapScanner.h"
+
+#include <Util/Alloc.h>
+#include <Util/Log.h>
+#include <Util/Utlist.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+// Helper structure for sorting
+typedef struct map_entry {
+    char* name;
+    struct map_entry *next, *prev;
+} map_entry_t;
+
+static int
+is_directory(const char* path)
```

Comment:
Follow the set coding style in general please. Format it as the rest of the code base

Reply:
Done !